extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -127,14 +127,13 @@ export async function getPublishDatabaseSettings(project: ISqlProject, promptFor
 	// if a publish profile was loaded and had a database name, use that instead of the project file name
 	const dbName = publishProfile?.databaseName || project.projectFileName;
 
-	// Ensure the project name or name specified in the publish profile is an option, either adding it if it
-	// doesn't already exist or moving it to the top if it does
-	const projectNameIndex = dbs.findIndex(db => db === dbName);
-	if (projectNameIndex === -1) {
+	// Ensure the project name or name specified in the publish profile is an option
+	const projectNameIndex = dbQuickpicks.findIndex(db => db.label === dbName);
```

Comment:
Changing this line from `dbs` to `dbQuickpicks` was the fix.  When a database with the proposed name already exists on the server, the earlier insertion of "+ Create New" to `dbQuickPicks` above threw the _actual_ index off by one when finding within the original db list.